contracts/root/lib/ChildChainVerifier.sol

```diff
@@ -0,0 +1,167 @@
+pragma solidity ^0.5.2;
+
+import { RLPReader } from "solidity-rlp/contracts/RLPReader.sol";
+import { BytesLib } from "../../common/lib/BytesLib.sol";
+import { Common } from "../../common/lib/Common.sol";
+import { RLPEncode } from "../../common/lib/RLPEncode.sol";
+import { Common } from "../../common/lib/Common.sol";
+import { MerklePatriciaProof } from "../../common/lib/MerklePatriciaProof.sol";
+
+import { Registry } from "../../common/Registry.sol";
+
+library ChildChainVerifier {
+  using RLPReader for bytes;
+  using RLPReader for RLPReader.RLPItem;
+
+  function processBurnReceipt(
+    bytes memory receiptBytes, bytes memory path, bytes memory receiptProof,
+    bytes32 receiptRoot, address sender)
+    internal
+    pure
+    returns (address rootToken, address childToken, uint256 amountOrTokenId)
+  {
+    RLPReader.RLPItem[] memory items = receiptBytes.toRlpItem().toList();
+    require(items.length == 4, "MALFORMED_RECEIPT");
+    // Do any other fields other than items[3] need to be checked?
+
+    // [3][1] -> [childTokenAddress, [WITHDRAW_EVENT_SIGNATURE, rootTokenAddress, sender], amount]
+    items = items[3].toList()[1].toList();
+    require(items.length == 3, "MALFORMED_RECEIPT"); // find a better msg
+    childToken = items[0].toAddress();
+    amountOrTokenId = items[2].toUint();
+
+    // [3][1][1] -> [WITHDRAW_EVENT_SIGNATURE, rootTokenAddress, sender]
+    items = items[1].toList();
+    require(items.length == 3, "MALFORMED_RECEIPT"); // find a better msg;
+    require(
+      // keccak256('Withdraw(address,address,uint256,uint256,uint256)') = 0xebff2602b3f468259e1e99f613fed6691f3a6526effe6ef3e768ba7ae7a36c4f
+      bytes32(items[0].toUint()) == 0xebff2602b3f468259e1e99f613fed6691f3a6526effe6ef3e768ba7ae7a36c4f,
+      "WITHDRAW_EVENT_SIGNATURE_NOT_FOUND"
+    );
+
```

Comment:
TODO: reduce these operations just to typecast bytes32 to address. we are doing a lot of extra ops.

Reply:
adding todo will do, no need to make changes will do later.